### PR TITLE
[Release 18] Update versions and links

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)

* Ran 
```
python utils/validate_versions.py --python-version=0.27.0 --csharp-version=2.1.0 --csharp-extensions-version=0.5.0 --release-tag=release_18
```
* Ran `pre-commit run validate-release-links-py --all-files` to update doc links to use the new tag
* Removed the `Unreleased` section in the changelog
* Updated the colab links to point to this release's doc tags, so that they'll be updated in the future (this is a one-time thing, no need to add to the release process)
* Fix bug in `utils/validate_release_links.py` that caused `None` in pip commands.
* Fix links in colab notebook to use `release_N_docs` instead of `release_N`

### Types of change(s)
- [x] Documentation update

### Checklist
- [x] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/main/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [x] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/main/docs) (if applicable)

### CODE REVIEW

File: colab/Colab_UnityEnvironment_1_Run.ipynb [hunk 1]
```diff
@@ -32,7 +32,7 @@
    },
    "source": [
     "# ML-Agents Open a UnityEnvironment\n",
-    "<img src=\"https://github.com/Unity-Technologies/ml-agents/blob/release_1/docs/images/image-banner.png?raw=true\" align=\"middle\" width=\"435\"/>"
+    "<img src=\"https://github.com/Unity-Technologies/ml-agents/blob/release_18/docs/images/image-banner.png?raw=true\" align=\"middle\" width=\"435\"/>"
```
Comment (chriselion) [inline colab/Colab_UnityEnvironment_1_Run.ipynb:35]: TODO make sure this image links still work. May need to move the images to a more permanent location.

File: docs/Installation-Anaconda-Windows.md [hunk 2]
```diff
@@ -151,7 +151,7 @@ config files in this directory when running `mlagents-learn`. Make sure you are
 connected to the Internet and then type in the Anaconda Prompt:
 
 ```console
-python -m pip install mlagents==0.26.0
+python -m pip install Nonemlagents==0.27.0
```
Comment (chriselion) [inline docs/Installation-Anaconda-Windows.md:154]: This is a bug, will fix

File: utils/validate_release_links.py [hunk 3]
```diff
@@ -85,7 +85,7 @@ def update_pip_install_line(line, package_verion):
     match = PIP_INSTALL_PATTERN.search(line)
     if match is not None:  # if there is a pip install line
         package_name = match.group("package")
-        quiet_option = match.group("quiet")
+        quiet_option = match.group("quiet") or ""
```
Comment (chriselion) [inline utils/validate_release_links.py:88]: Maybe there's a better way to do this with the capture in the regex, but I think this works too.

File: colab/Colab_UnityEnvironment_1_Run.ipynb (hunk 1 above)
Comment (chriselion) [reply colab/Colab_UnityEnvironment_1_Run.ipynb:35]: Broken, will fix.

**Review by dongruoping — APPROVED** [review]

**Pull request merged**